optsr.py

```diff
@@ -0,0 +1,280 @@
+import numpy as np
```

Comment:
Please remove this file from the commit; it looks like it was accidentally included.